upup/pkg/api/cluster.go

```diff
@@ -240,11 +240,13 @@ type EtcdClusterSpec struct {
 
 type EtcdMemberSpec struct {
 	// Name is the name of the member within the etcd cluster
-	Name string `json:"name,omitempty"`
-	Zone string `json:"zone,omitempty"`
+	Name            string `json:"name,omitempty"`
+	Zone            string `json:"zone,omitempty"`
 
-	VolumeType string `json:"volumeType,omitempty"`
-	VolumeSize int    `json:"volumeSize,omitempty"`
+	VolumeType      string `json:"volumeType,omitempty"`
+	VolumeSize      int    `json:"volumeSize,omitempty"`
+	KmsKeyId        string `json:"kmsKeyId,omitempty"`
+	EncryptedVolume bool   `json:"encryptedVolume,omitempty"`
```

Comment:
Historically we didn't do this, and we are suffering for it.  For all new
APIs and API changes, all optional fields must be pointers.  Please take
this message and spread the word.  It's not practical for every API review
to go through Brian and I, so sometimes it gets missed.  For better or for
worse, we're using Go as our IDL, for now.



On Wed, Sep 7, 2016 at 10:11 AM, yissachar <notifications@github.com> wrote:

> In upup/pkg/api/cluster.go
> <https://github.com/kubernetes/kops/pull/398#discussion_r77865316>:
>
> >
> > -	VolumeType string `json:"volumeType,omitempty"`
> > -	VolumeSize int    `json:"volumeSize,omitempty"`
> > +	VolumeType      string `json:"volumeType,omitempty"`
> > +	VolumeSize      int    `json:"volumeSize,omitempty"`
> > +	KmsKeyId        string `json:"kmsKeyId,omitempty"`
> > +	EncryptedVolume bool   `json:"encryptedVolume,omitempty"`
>
> So should KmsKeyId be *string? That is what it sounds like from your
> comment, but doesn't seem to jive with the existing code. For instance
> VolumeType is string and VolumeSize is int, even though both of those are
> optional as far as I can tell.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kops/pull/398/files/8c3b2274d91e1c7e3994eeb4cc5450f17fd966d4#r77865316>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFVgVFRX3OQDf0qPge0a_O6zfR5xk9MIks5qnvAzgaJpZM4JzDC_>
> .
>